src/wristpy/processing/metrics.py

```diff
@@ -53,3 +54,184 @@ def angle_relative_to_horizontal(
     angle_degrees = np.degrees(angle_radians)
 
     return models.Measurement(measurements=angle_degrees, time=acceleration.time)
+
+
+def detect_nonwear(
+    acceleration: models.Measurement,
+    short_epoch_length: int = 900,
+    long_epoch_length: int = 3600,
+    std_criteria: float = 0.013,
+    range_criteria: float = 0.05,
+) -> models.Measurement:
+    """Set non_wear_flag based on accelerometer data.
+
+    This implements GGIR "2023" non-wear detection algorithm.
+    Briefly, the algorithm, creates a sliding window of long epoch length that steps
+    forward by the short epoch length.
+    It checks if the acceleration data in that long window, for each axis, meets certain
+    criteria thresholds to compute a non-wear value.
+    And then applies that non-wear value to all the short windows that make up the long
+    window. Additionally, for overlapping windows, the maximum of the non-wear value for
+    the overlaps is kept. Finally, there is a pass to find isolated "1s" in the non-wear
+    value, and set them to 2 if surrounded by > 1 values. The non-wear flag is set to
+    1 (true) if the non-wear value is >= 2, and 0 (false) otherwise.
+
+
+    Args:
+        acceleration: The Measurment instance that contains the calibrated acceleration
+            data.
+        short_epoch_length: The short window size, in seconds, for non-wear detection
+        long_epoch_length: The long window size, in seconds, for non-wear detection.
+            Ideally, it should be a multiple of short_epoch_length.
+        std_criteria: Threshold criteria for standard deviation
+        range_criteria: Threshold criteria for range of acceleration
+
+
+    Returns:
+        A new Measurment instance with the non-wear flag and corresponding timestamps.
+    """
+    acceleration_grouped_by_short_window = _group_acceleration_data_by_time(
+        acceleration, short_epoch_length
+    )
+
+    n_short_epoch_in_long_epoch = int(long_epoch_length / short_epoch_length)
```

Comment:
@ReinderVosDeWael thoughts on handling this. I can easily add a check/ Raise a ValueError if it isn't a multiple of short_epoch_length

Reply:
Is the result for short not exactly divisible by long valid? If it's valid, then why do we say it is ideally a multiple? If it's not valid, why do we allow it?

Reply:
In GGIR, it is hardcoded to such that the short window is 900s and the long window is 3600s (the default values I set). Currently, it is valid and functional for any long_window that is a multiple of short_window. In theory, it could work with non-multiples? But there could be some issues at the end of the acceleration signal and I haven't tested the validity of this.

Reply:
Can we use an integer like `n_sub_windows` to compute the length of the 'short window' to side-step the user having to do math?

Reply:
I am going to add this (and the equivalent test) for now. But maybe I will make a Task for future investigation

```   
 if long_epoch_length % short_epoch_length != 0:
        raise ValueError(
            "long_epoch_length must be an integer multiple of short_epoch_length"
        )
```

Reply:
I don't know why I don't see these comments right away... but yeah that makes sense. What about the following modification:

We have the following `short_window_length: int` and `n_short_windows_in_long_epoch:int` instead of `long_epoch_length`. This way we don't need to worry about odd or prime `long_epoch_length`. We only use `long_epoch_length` to get the number of short windows that make up the long window.